kernel/src/device/evdev/file.rs

```
@@ -185,6 +214,192 @@ impl EvdevFile {
+                _ => None,
+            },
+            _ => None,
+        }
+    }
+
+    fn handle_evdev_ioctl(&self, raw: u32, arg: usize) -> Result<()> {
+        match Self::parse_evdev(raw) {
+            Some(EvdevIoctl::GetName { len }) => {
+                let evdev = self.upgrade_evdev_device()?;
+                self.write_string_to_userspace(evdev.device.name(), len as usize, arg)?;
+            }
+            Some(EvdevIoctl::GetPhys { len }) => {
+                let evdev = self.upgrade_evdev_device()?;
+                self.write_string_to_userspace(evdev.device.phys(), len as usize, arg)?;
+            }
+            Some(EvdevIoctl::GetUniq { len }) => {
+                let evdev = self.upgrade_evdev_device()?;
+                self.write_string_to_userspace(evdev.device.uniq(), len as usize, arg)?;
+            }
+            Some(EvdevIoctl::GetId) => {
+                let evdev = self.upgrade_evdev_device()?;
+                let id = evdev.device.id();
+                current_userspace!().write_val(arg, &id)?;
+            }
+            Some(EvdevIoctl::GetVersion) => {
+                current_userspace!().write_val(arg, &EVDEV_DRIVER_VERSION)?;
+            }
+            Some(EvdevIoctl::GetBit { event_type, len }) => {
+                let evdev = self.upgrade_evdev_device()?;
+                let capability = evdev.device.capability();
+                let event_types_bytes = capability.event_types_bits().to_le_bytes();
+                let bitmap = match event_type as u16 {
+                    0 => Some(&event_types_bytes[..]),
+                    t if t == EventTypes::KEY.as_index() => {
+                        Some(capability.supported_keys_bitmap())
+                    }
+                    t if t == EventTypes::REL.as_index() => {
+                        Some(capability.supported_relative_axes_bitmap())
+                    }
+                    _ => None,
```

If `t` is not a recognized type, shouldn't we return an error?